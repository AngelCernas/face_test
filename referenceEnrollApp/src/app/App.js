import React from 'react';
import { StyleSheet } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator, StackNavigationOptions } from '@react-navigation/stack';
import ImageCapture from '../features/screens/ImageCapture';
import Login from '../features/screens/Login';
import Welcome from '../features/screens/Welcome';
import Consent from '../features/screens/Consent';
import Instruction from '../features/screens/Instruction';
import Receipt from '../features/screens/Receipt';
import ManageProfile from '../features/screens/ManageProfile';
import Settings from '../features/screens/Settings';
import { Provider } from 'react-redux';
import configureStore from './store';
const RNFS = require('react-native-fs');
import * as constants from '../shared/constants';


const Stack = createStackNavigator();
const store = configureStore();

const App = () => {
  /*
  To store username and personId information, this app writes the data
  to the enrollment directory created here. This is ONLY for demonstration purposes. 
  Any user information and personId should be stored in a secured, encrypted database. 
  A user's personId should be treated as a secret.
  */
  RNFS.mkdir(RNFS.DocumentDirectoryPath + '/enrollment/');

  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            options={{ headerShown: false, gestureEnabled: false }}
            name={constants.SCREENS.welcome}
            component={Welcome}
          />

          <Stack.Screen
            options={{ headerShown: false, gestureEnabled: false }}
            name={'Settings'}
            component={Settings}
          />

          <Stack.Screen
            options={{ title: '', headerStyle: styles.header, gestureEnabled: false }}
            name={constants.SCREENS.manage}
            component={ManageProfile}
          />
          <Stack.Screen
            options={{
              title: '',
              headerStyle: styles.header, gestureEnabled: false
            }}
            name={constants.SCREENS.consent}
            component={Consent}
          />
          <Stack.Screen
            options={{ title: '', headerStyle: styles.header, gestureEnabled: false }}
            name={constants.SCREENS.login}
            component={Login}
          />
          <Stack.Screen
            options={{ title: '', headerStyle: styles.header, gestureEnabled: false }}
            name={constants.SCREENS.instruction}
            component={Instruction}
          />
          <Stack.Screen
            options={{ headerShown: false, gestureEnabled: false }}
            name={constants.SCREENS.imageCapture}
            component={ImageCapture}
          />
          <Stack.Screen
            options={{ title: '', headerStyle: styles.header, gestureEnabled: false }}
            name={constants.SCREENS.receipt}
            component={Receipt}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

var styles = StyleSheet.create({
  header: {
    backgroundColor: '#0078D4',
  },
});

export default App;
