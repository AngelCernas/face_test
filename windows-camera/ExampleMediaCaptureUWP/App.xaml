<!--

Copyright (c) Microsoft Corporation. All rights reserved.

This code is licensed under the MIT License (MIT).
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

-->
<Application
    x:Class="ExampleMediaCaptureUWP.App"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:ExampleMediaCaptureUWP">
    <Application.Resources>
        <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls"/>
    </Application.Resources>
</Application>
