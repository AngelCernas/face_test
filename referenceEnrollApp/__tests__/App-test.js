/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/app/App';
// Note: test renderer must be required after react-native.
import {act, create} from 'react-test-renderer';

// Silences warnings
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

it('renders correctly', async () => {
  // render the app
  const result = create(<App />);
  await act(async () => {
    expect(result).toMatchSnapshot();
  });
});
