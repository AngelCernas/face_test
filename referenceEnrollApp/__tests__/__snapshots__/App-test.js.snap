// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#002C55",
                          "flex": 1,
                          "flexDirection": "row",
                          "justifyContent": "center",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "flexDirection": "column",
                          }
                        }
                      >
                        <View
                          accessibilityIgnoresInvertColors={true}
                          style={
                            Object {
                              "backgroundColor": "#002C55",
                              "flex": 1,
                              "justifyContent": "center",
                              "resizeMode": "cover",
                              "width": "100%",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../src/assets/bg_heroIllustration_welcome.png",
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "bottom": 0,
                                  "left": 0,
                                  "position": "absolute",
                                  "right": 0,
                                  "top": 0,
                                },
                                Object {
                                  "height": undefined,
                                  "width": "100%",
                                },
                                undefined,
                              ]
                            }
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "flex-end",
                            "bottom": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "backgroundColor": "white",
                                "borderRadius": 4,
                                "flex": 1,
                                "flexDirection": "column",
                                "margin": 16,
                                "padding": 15,
                              },
                              Object {
                                "height": "50%",
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "justifyContent": "flex-start",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#77787B",
                                  "fontFamily": "Roboto",
                                  "fontSize": 12,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                  "lineHeight": 16,
                                }
                              }
                            >
                              Contoso | Real Estate & Security
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "marginBottom": 20,
                                "marginTop": 15,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "marginBottom": 10,
                                }
                              }
                            >
                              <Text
                                style={
                                  Object {
                                    "fontFamily": "Roboto",
                                    "fontSize": 24,
                                    "fontStyle": "normal",
                                    "fontWeight": "bold",
                                    "lineHeight": 28,
                                  }
                                }
                              >
                                An easier way to get into work
                              </Text>
                            </View>
                            <Text
                              style={
                                Object {
                                  "color": "#77787B",
                                  "fontFamily": "Roboto",
                                  "fontSize": 16,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                  "lineHeight": 22,
                                }
                              }
                            >
                              You can now use face recognition instead of your badge to conveniently unlock building doors.
                            </Text>
                          </View>
                          <View
                            style={
                              Object {
                                "color": "red",
                                "paddingBottom": 20,
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "#0078D4",
                                      "borderRadius": 4,
                                      "paddingHorizontal": 15,
                                      "paddingVertical": 15,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontFamily": "Roboto",
                                      "fontSize": 16,
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                      "lineHeight": 22,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "fontWeight": "bold",
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "color": "white",
                                          },
                                          undefined,
                                        ]
                                      }
                                    >
                                      Get started
                                    </Text>
                                  </Text>
                                </Text>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "color": "red",
                                "paddingBottom": 20,
                              }
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "#FFFFFF",
                                      "borderColor": "#0078D4",
                                      "borderRadius": 4,
                                      "borderWidth": 1,
                                      "paddingHorizontal": 15,
                                      "paddingVertical": 15,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "fontFamily": "Roboto",
                                      "fontSize": 16,
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                      "lineHeight": 22,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "fontWeight": "bold",
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Object {
                                            "color": "#0078D4",
                                          },
                                          undefined,
                                        ]
                                      }
                                    >
                                      Manage profile
                                    </Text>
                                  </Text>
                                </Text>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "justifyContent": "flex-end",
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#77787B",
                                  "fontFamily": "Roboto",
                                  "fontSize": 12,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                  "lineHeight": 16,
                                }
                              }
                            >
                              Details at contoso.com/touchless-access
                              


                              Contoso Privacy Statement
                            </Text>
                          </View>
                        </View>
                        <View />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
